examples/3d/generate_custom_mesh.rs

```diff
@@ -0,0 +1,238 @@
+        .run();
+}
+
+fn setup(
+    mut commands: Commands,
+    asset_server: ResMut<AssetServer>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+) {
+    // Importing the custom texture.
+    let custom_texture_handle: Handle<Image> = asset_server.load("textures/array_texture.png");
+    // Creating and saving a handle to the mesh.
+    let cube_mesh_handle: Handle<Mesh> = meshes.add(create_cube_mesh());
+
+    // Rendering the mesh with the custom texture using a PbrBundle.
+    commands.spawn(PbrBundle {
+        mesh: cube_mesh_handle,
+        material: materials.add(StandardMaterial {
+            base_color_texture: Some(custom_texture_handle),
+            ..default()
+        }),
+        ..default()
+    });
+
+    // Transform for the camera and lighting, looking at (0,0,0) (the position of the mesh).
+    let camera_and_light_transform =
+        Transform::from_xyz(2.1, 2.1, 2.1).looking_at(Vec3::ZERO, Vec3::Y);
+
+    // Camera in 3D space.
+    commands.spawn(Camera3dBundle {
+        transform: camera_and_light_transform,
+        ..default()
+    });
+
+    // Lighting up the scene.
+    commands.spawn(PointLightBundle {
+        point_light: PointLight {
+            intensity: 1000.0,
+            range: 100.0,
+            ..default()
+        },
+        transform: camera_and_light_transform,
+        ..default()
+    });
+}
+
+// System to recieve input from the user, check out examples/input/ for more examples about user
+// input.
+fn input_handler(
+    keyboard_input: Res<Input<KeyCode>>,
+    mesh_query: Query<&Handle<Mesh>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut query: Query<&mut Transform, With<Handle<Mesh>>>,
+    time: Res<Time>,
+) {
+    if keyboard_input.just_pressed(KeyCode::Space) {
+        let mesh_handle = mesh_query.get_single().expect("Query not successful");
+        let mesh = meshes.get_mut(mesh_handle).unwrap();
+        toggle_texture(mesh);
+    }
+    if keyboard_input.pressed(KeyCode::X) {
+        for mut transform in &mut query {
+            transform.rotate_x(time.delta_seconds() / 1.2);
+        }
+    }
+    if keyboard_input.pressed(KeyCode::Y) {
+        for mut transform in &mut query {
+            transform.rotate_y(time.delta_seconds() / 1.2);
+        }
+    }
+    if keyboard_input.pressed(KeyCode::Z) {
+        for mut transform in &mut query {
+            transform.rotate_z(time.delta_seconds() / 1.2);
+        }
+    }
+}
+
+fn create_cube_mesh() -> Mesh {
+    let mut cube_mesh = Mesh::new(PrimitiveTopology::TriangleList);
+
+    #[rustfmt::skip]
+    cube_mesh.insert_attribute(
+        Mesh::ATTRIBUTE_POSITION,
+        // Each array is an [x, y, z] coordinate in local space.
+        vec![
+            // top (facing towards +y)
+            [0.0, 1.0, 0.0], // vertex with index 0
+            [1.0, 1.0, 0.0], // vertex wtih index 1
+            [1.0, 1.0, 1.0], // etc. until 23
+            [0.0, 1.0, 1.0],
```

Comment:
Since the origin is at the bottom left, this behaves differently from a bevy `shape::Cube`.

The bevy cube has the vertices at the -0.5 and 0.5 coordinates, so it is fully centered and when rotating it, it rotates around the center.

This cube's center is one of its vertices, and the rotation doesn't act the same way.

I suggest writing a comment pointing that out.

Reply:
I'll just change it, better to have it centred on 0,0,0 anyway so rotation looks better.